src/daft-sql/src/functions.rs

```diff
@@ -51,50 +51,43 @@ fn check_features(func: &Function) -> SQLPlannerResult<()> {
 }
 
 /// [SQLFunction] extends [FunctionExpr] with basic input validation (arity-check).
-///
-/// TODOs
-///  - Support for ScalarUDF as either another enum variant or make SQLFunction a trait.
-pub enum SQLFunction {
-    Function(FunctionExpr),
-    Agg(AggExpr),
+pub trait SQLFunction: Send + Sync {
+    /// helper function to extract the function args into a list of [ExprRef]
+    /// This is used to convert unnamed arguments into [ExprRef]s.
+    /// ```sql
+    /// SELECT concat('hello', 'world');
+    /// ```
+    /// this  will be converted to: [lit("hello"), lit("world")]
```

Comment:
Do we need to differentiate between function args that take as. input expressions vs functions that mandate that we can only take only take in scalar inputs?

Reply:
I'm not sure I fully understand the question. 

are you asking if function args can accept exprs as inputs as well? 
such as `some_function(column1, arg0 := column2)`?

Reply:
Or another example is the discussion we had a few months ago regarding `seed`?